xarray/core/common.py

```diff
@@ -868,3 +868,31 @@ def ones_like(other, dtype=None):
     """Shorthand for full_like(other, 1, dtype)
     """
     return full_like(other, 1, dtype)
+
+
+def is_np_datetime_like(dtype):
+    """Check if a dtype is a subclass of the numpy datetime types
+    """
+    return (np.issubdtype(dtype, np.datetime64) or
+            np.issubdtype(dtype, np.timedelta64))
+
+
+def contains_cftime_datetimes(var):
+    """Check if a variable contains cftime datetime objects"""
+    try:
+        from cftime import datetime as cftime_datetime
+    except ImportError:
+        return False
+    else:
+        sample = var.data.ravel()[0]
```

Comment:
Since this could be potentially called on many arrays, let's be a little more careful before calculating `sample`:
1. Let's verify that the array has `dtype=object` (otherwise it can't contain cftime.datetime objects)
2. Let's verify that the array has `size > 0` before trying any elements.